test/routes/api/tasks/tasks.test.ts

```diff
@@ -509,6 +513,31 @@ describe('Tasks api (logged user only)', () => {
         assert.strictEqual(message, 'File uploaded successfully')
       })
 
+      it('should delete existing image for a task before uploading a new one', async (t) => {
+        app = await build(t)
+
+        await uploadImageForTask(app, taskId, testImagePath, uploadDirTask)
+
+        const { mock: mockUnlink } = t.mock.method(fs.promises, 'unlink')
+
+        const form = new FormData()
+        form.append('file', fs.createReadStream(testImagePath))
+
+        const res = await app.injectWithLogin('basic', {
+          method: 'POST',
+          url: `/api/tasks/${taskId}/upload`,
+          payload: form,
+          headers: form.getHeaders()
+        })
+
+        assert.strictEqual(res.statusCode, 200)
+
+        const { message } = JSON.parse(res.payload)
+        assert.strictEqual(message, 'File uploaded successfully')
+
+        assert.strictEqual(mockUnlink.callCount(), 1)
```

Comment:
A safer way to test this would be to upload the first time with a different image (so different filename):
```ts
await uploadImageForTask(
  app, 
  taskId, 
  imageToReplacePath,  // Here...
  uploadDirTask
 )
```

And check after the request that the file is not present in the upload dir anymore.